go-controller/pkg/ovn/logical_switch_manager.go

```diff
@@ -315,7 +315,14 @@ func initJoinLogicalSwitchIPManager() (*joinSwitchIPManager, error) {
 		lrpIPCache: make(map[string][]*net.IPNet),
 	}
 	var joinSubnets []*net.IPNet
-	for _, joinSubnetString := range []string{config.Gateway.V4JoinSubnet, config.Gateway.V6JoinSubnet} {
+	joinSubnetsConfig := []string{}
+	if config.IPv4Mode {
```

Comment:
I'm not sure we can do this. I'm wondering if there is any repercussions if previously the join subnet was always dualstack and then we try to upgrade. @aojea fyi for dualstack conversion stuff

Is this required for this PR to work?

Reply:
we need this for the new unit-test to work short of making address set changes. because the address set for v6 needs to be created for this namespace and if the config.IPv6Mode is not set, the v6 address set is not created.

Reply:
we can't break things so often, there is a comment in this function

> // Initializes a new join switch logical switch manager.
> // This IPmanager guaranteed to always have both IPv4 and IPv6 regardless of dual-stack

not able to go deep in this PR, but why can't the test work?

Reply:
When we create the address sets for the namespace, the address set factory creation logic checks if config.IPV6Mode is set. The unit-test is trying to validate that both v4 and v6 join IPs are present in the corresponding address sets and fails.

Reply:
For my edification, why do we need to have v6 join IPs if the deployment is single-stack?

Reply:
I think that is because of simplification ... there are some ovn commands that can't be dualstack and you need to delete and recreate causing disruption

Reply:
Yeah, having the v6 join IP there generally doesn't hurt, and it makes it (much) simpler to do single-to-dual migration.

Reply:
@trozet and I went through the code for `joinSubnets` and what happens when `syncNodes` is called. It seems like the only side-effect of doing this for upgrades is that the nbdb will have a unused v6 LRP IP in the cache. For single to dual-stack conversion, this likely wouldn't pose a problem.

Reply:
I think it is OK for upgrade and it passed dualstack conversion anyway :+1: